### PR TITLE
Make all tests run independently

### PR DESCRIPTION
**Pull request by ppannuto**
**Note: Depends on PR #162**

### CODE REVIEW

**Comment by ppannuto** [comment]
Oops, j/k only ran the tests I added locally. Will fix, one moment.

**Comment by ppannuto** [comment]
Fixed. Sorry, first push was on the way out.

**Comment by dandv** [comment]
@connor4312: any chance to pull in more tests PRs? Would be great for my work in #177.

**Comment by dandv** [comment]
@connor4312 - [same comment](https://github.com/node-influx/node-influx/pull/164#issuecomment-247486363) as for #164 :)

**Comment by ppannuto** [comment]
FWIW, this is probably still worth considering in addition to #183. 

The big contribution of this PR is that it allows the tests to run independently instead of depending on each other, which is allows you to isolate issues to exactly one test case (and prevents accidental assumptions from slipping in due to undocumented inter-test dependencies) and allows you to spin rapidly on a test that fails rather than spinning the whole suite. 

A poorly titled PR by the time I was done with it, sorry.

**Comment by dandv** [comment]
BTW @connor4312, I found upgrading to [node v6.6.0](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V6.md#6.6.0) really useful. A bunch of promise rejections are unhandled, and Node now warns about them.

File: index.js [hunk 1]
```diff
@@ -76,13 +76,17 @@ var InfluxDB = function (options) {
...
   var results = []
   _.each(response, function (result) {
     var tmp = []
     if (result.series) {
       _.each(result.series, function (series) {
         var rows = _.map(series.values, function (values) {
-          return _.extend(_.zipObject(series.columns, values), series.tags)
+          var columnsDesanitized = _.map(series.columns, self._desanitizeString)
+          var valuesDesanitized = _.map(values, self._desanitizeString)
+          var tagsDesanitized = _.forOwn(series.tags, self._desanitizeObjectIter.bind(self))
```
Comment (connor4312) [inline index.js]: I'm not sure this is needed. In 0.13.0 it doesn't look like results of queries include the original escaping characters:

```js
> insert test_series,tag1=value\ b,tag\ 2=x value=2
> select * from test_series
{
    "results": [
        {
            "series": [
                {
                    "name": "test_series",
                    "columns": [
                        "time",
                        "tag 2",
                        "tag1",
                        "value"
                    ],
                    "values": [
                        [
                            1474519055673092238,
                            "x",
                            "value b",
                            2
                        ]
                    ]
                }
            ]
        }
    ]
}
```

**Comment by connor4312** [comment]
@dandv hm, I upgraded but am not getting any warnings in the tests...

**Comment by connor4312** [comment]
Merged! I remove the escaping since it seemed like @dandv's earlier PR covered that better, and unescaping in _parseResults does not seem necessary (see the `#escapingRoundtrip` test).

**Comment by dandv** [comment]
Thanks @connor4312! I wanted to make the same comment about unescaping/desanitizing being unnecessary.

File: index.js (hunk 1 above)
Comment (dandv) [inline index.js]: Where is this output from? The [JSON line protocol was deprecated](https://docs.influxdata.com/influxdb/v0.13/write_protocols/json/).

Comment (connor4312) [inline index.js]: This is via the Influx CLI, the results are from [`/query`](https://docs.influxdata.com/influxdb/v1.0/guides/querying_data/) API. This output format can be toggled like:

```
➜  ~ influx
Visit https://enterprise.influxdata.com to register for updates, InfluxDB server management, and monitoring.
Connected to http://localhost:8086 version v1.0.0
InfluxDB shell version: v1.0.0
> format json
> pretty
Pretty print enabled
> // all query outputs will be json now
```

**Comment by dandv** [comment]
> @dandv hm, I upgraded but am not getting any warnings in the tests...

Nevermind, they were caused by my work on `request` replacements.

**Pull request merged**